### DISCUSSION TITLE
useActionData on layout after v2.3.0

### DISCUSSION THREAD

**Discussion by eduardodallmann** [remix-run/remix#8188] [Q&A]
Hello, after version `2.3.0` the `useActionData` hook does not trigger the reaction.

I have a layout route `_app/route.tsx` and a route with a form `_app/document.tsx`, I also have a Form component. All 3 use `useActionData` and will show an alert when the action data changes. But the layout route doesn't react. If I change to version 2.2.0 everything works again.
I saw in the changelog that some things changed, I tried to include `unstable_flushSync` both `true` and `false` but without success. What am I missing?

Example:
https://codesandbox.io/p/devbox/aged-cache-jw7k7s

**Comment by eduardodallmann** [comment]
Looking at `react-router`, I believe the change that impacted was this
https://github.com/remix-run/react-router/issues/11022

Contextualizing, in my current project, I use the `useActionData` hook in a layout route to always display the same error modal if the action of any child route returns `{globalError: true, message: 'blabla'}`.
If I can no longer use `useActionData` for this, what would be a good alternative?

**Reply by hi-ogawa** [reply]
Hey, there are some comments in the linked PR. You might find these helpful https://github.com/remix-run/react-router/pull/11023#issuecomment-1828396713 https://github.com/remix-run/react-router/pull/11023#issuecomment-1826204148

**Reply by eduardodallmann** [reply]
thanks @hi-ogawa your suggestion works like a charm, although I'm afraid to use UNSAFE 😬

https://codesandbox.io/p/devbox/nostalgic-http-7zs7rp

**Reply by kiliman** [reply]
NOTE: the `unsafe_` prefix simply means the API may change in the future. It doesn't mean it's buggy or will cause issues in your app. The Remix team wants people to use these APIs to ensure they cover the proper use cases. Once the requirements are fully understood, they will update the API and remove the `unsafe_` prefix.
